### PR TITLE
foreign asset creator pallet

### PR DESCRIPTION
**Pull request by girazoki**
This PR proposes a pallet that is similar to the asset-manager from moonbeam but it is a simplified version of it. In this case, the foreign-asset-creator-pallet is only responsible for creating and maintaining an `assetId`<->`ForeignAsset` mapping to ease the following situations:

- knowing which `assetId` to touch whenever we receive an XCM `MultiAsset`
- knowing which `MultiAsset` to send when we are told to send a specific `assetId` to another chain

Important changes:

- **We allow the creator to decide an `assetId` instead of deriving it ourselves through some other mechanism.** The main reason for having a custom derivation for the `assetId` was that one could find a collision with a smart contract on the precompile address, but I believe this case is very ethereum specific and could be addressed in different manners (for instance, by using a smaller `assetId` type)

- No units per second, this if needed should be done in a different pallet

- No local asset creation

### CODE REVIEW

File: pallets/foreign-asset-creator/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	"parity-scale-codec/std",
+	"scale-info/std",
+	"serde",
+	"sp-io/std",
+	"sp-runtime/std",
+	"sp-std/std"
+]
+
+runtime-benchmarks = [ "frame-benchmarking" ]
+try-runtime = [ "frame-support/try-runtime" ]
```
Comment (librelois) [inline pallets/foreign-asset-creator/Cargo.toml:43]: Now try-runtime is a separate binary, so we don't need this rust feature anymore

**Review by librelois — APPROVED** [review]

**Comment by librelois** [comment]
@girazoki can you merge master?

**Comment by girazoki** [comment]
will do yes

**Comment by girazoki** [comment]
Done @librelois but I still want to push some benchmarking code

**Pull request merged**